addons/mail/models/mail_message.py

```diff
@@ -918,6 +930,7 @@ def _message_format(self, fnames, format_reply=True):
                 'record_name': record_name,
                 'create_date': message_sudo.create_date,
                 'write_date': message_sudo.write_date,
+                'canReply': message_post_rights_by_model_name[message_sudo.model] if message_sudo.model else False,
```

Comment:
Instead of `canReply`, could explicit name and provide the read/write access of thread.
That way you don't have to introduce a new concept, and in JS you don't have to `fetchData()` for chat windows, as fetching messages will have the necessary data to determine whether you have access rights on thread to post messages.